### PR TITLE
[Model] Create QuoteStatus Model

### PR DESCRIPTION
**Pull request by vinibr3**
Create Model QuoteStatus and his tests.

### CODE REVIEW

File: spec/factories.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :quote_status do
+    name "Em analise"
```
Comment (kaiomagalhaes) [inline spec/factories.rb]: We should avoid strings in portuguese

File: app/models/quote_status.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class QuoteStatus < ApplicationRecord
+  validates :name, length: { maximum: 255 }, presence: true, uniqueness: true
+  validates :order, length: { maximum: 255 }, numericality: { integer: true }, presence: true
```
Comment (kaiomagalhaes) [inline app/models/quote_status.rb:3]: this field is a string and you are adding a `numericality` to it, it doesn't make sense

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: app/models/quote_status.rb (hunk 2 above)
Comment (vinibr3) [reply app/models/quote_status.rb:3]: Order will not make a math operation. So it puts it as string. And also because the form will provide this value as a string, so it will have to convert to integer to save to the database

Comment (kaiomagalhaes) [reply app/models/quote_status.rb:3]: Okay, I will approve because I suppose that you guys already discussed the options here, but I STRONGLY suggest you to change it to use https://github.com/pluginaweek/state_machine. You guys are basically going to have to rewrite a lot of code to validate the status and to make sure it doesn't go from one state to an invalid state. This current approach is very prone errors and to be honest I'm sure that if you guys decide to add ANY kind of validations you will have to use this `order` in math operations. 

TL;DR: I'm approving because I'm only reviewing the code, but if I had to review the flow it would be reproved. 

@Heirian check this

**Review by kaiomagalhaes — DISMISSED** [review]

**Review by Heirian — DISMISSED** [review]

**Comment by Heirian** [comment]
Please, @vinibr3 fix the conflict

File: app/models/quote_status.rb (hunk 2 above)
Comment (Heirian) [reply app/models/quote_status.rb:3]: That was an architectural decision of @vinibr3 . They are free to decide whether they use it or not, but this kind of thing will be evaluated at the end of the project.

**Review by Heirian — DISMISSED** [review]

**Review by RodrigoNM — DISMISSED** [review]

File: spec/factories.rb [hunk 3]
```diff
@@ -1,4 +1,9 @@
 FactoryBot.define do
+  factory :quote_status do
+    name "In analysis"
```
Comment (kaiomagalhaes) [inline spec/factories.rb]: please use Faker here

File: spec/models/quote_status_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require "rails_helper"
+
+RSpec.describe QuoteStatus, type: :model do
+  it { should validate_presence_of(:name) }
+  it { should validate_length_of(:name).is_at_most(255) }
+  describe "validations" do
```
Comment (kaiomagalhaes) [inline spec/models/quote_status_spec.rb]: please add a line between `it` and `describe`

File: spec/models/quote_status_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    subject { quote_status = build(:quote_status) }
+    it { should validate_uniqueness_of(:name) }
+  end
+
+  it { should validate_presence_of(:order) }
```
Comment (kaiomagalhaes) [inline spec/models/quote_status_spec.rb]: please keep the `it` together

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: spec/models/quote_status_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  it { should validate_length_of(:order).is_at_most(255) }
```
Comment (kaiomagalhaes) [inline spec/models/quote_status_spec.rb]: please use `expect` instead of `should`

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

**Review by RodrigoNM — APPROVED** [review]

**Review by RodrigoNM — DISMISSED** [review]

**Review by RodrigoNM — DISMISSED** [review]

**Review by Heirian — DISMISSED** [review]

**Review by Heirian — APPROVED** [review]

**Review by RodrigoNM — APPROVED** [review]

**Review by kaiomagalhaes — APPROVED** [review]
